exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/UnionAllPrule.java

```diff
@@ -61,7 +61,8 @@ public void onMatch(RelOptRuleCall call) {
         convertedInputList.add(convertedInput);
       }
 
-      traits = call.getPlanner().emptyTraitSet().plus(Prel.DRILL_PHYSICAL).plus(DrillDistributionTrait.SINGLETON);
+      // distribution trait is set to ANY to allow Union-All to inherit the distribution of its child
+      traits = call.getPlanner().emptyTraitSet().plus(Prel.DRILL_PHYSICAL).plus(DrillDistributionTrait.ANY);
```

Comment:
Will it work by setting "ANY" in case inputs of union-all has different distribution trait?

Let's say LHS has Hash-distribution(col1), RHS has Hash-distribution(co1, col2), or Random distribution. What will union-all's distribution trait? Will it use LHS, or RHS ?